content/blog/improved-kubernetes-await-logic/index.md

```diff
@@ -0,0 +1,293 @@
+However, now we can use the `pulumi.com/waitFor` annotation to instruct Pulumi to wait until web hooks have CA data injected before proceeding to create Certificates.
+One way to achieve this is by using the new [transforms API](https://www.pulumi.com/blog/resource-transforms/) to annotate the web hooks appropriately:
+
+```typescript
+const install = new kubernetes.yaml.v2.ConfigFile("install",
+  {
+    file: "https://github.com/cert-manager/cert-manager/releases/download/v1.15.2/cert-manager.yaml",
+  },
+  {
+    // Add a "waitFor" annotation which waits for the cainjector pod to come
+    // online and modify our webhooks with valid CA bundles. Webhooks
+    // needed by our Certificate will fail until this happens.
+    transforms: [
+      (args) => {
+        if (
+          args.type ===
+            "kubernetes:admissionregistration.k8s.io/v1:ValidatingWebhookConfiguration" ||
+          args.type ===
+            "kubernetes:admissionregistration.k8s.io/v1:MutatingWebhookConfiguration"
+        ) {
+          args.props.metadata.annotations["pulumi.com/waitFor"] =
+            "jsonpath={.webhooks[*].clientConfig.caBundle}";
+          return {
+            props: args.props,
+            opts: args.opts,
+          };
+        }
+        return undefined;
+      },
+    ],
+  },
+);
+```
+
+Here we've given the `pulumi.com/waitFor` annotation a value of `jsonpath={.webhooks[*].clientConfig.caBundle}`, which is a [JSONPath] expression that will wait until a web hook in the `webhooks` array has a `clientConfig` with a `caBundle` populated -- indicating that the web hook is usable.
+
+Now the program will deploy successfully with a `Missing {.webhooks[*].clientConfig.caBundle}` message while Pulumi waits for the field to be configured.
+
+![cert-manager](cert-manager.png)
+
+## Experimental: Generic readiness
+
+Being able to specify custom readiness is powerful, but a lot of resources follow well-known conventions like the `Ready=True` condition.
+Rather than requiring you to specify `"pulumi.com/waitFor": "condition=Ready"` on every resource that needs it, there's now an opt-in feature to automatically apply well-known readiness criteria to all resources managed by Pulumi.
+
+Running your program with the environment variable `PULUMI_K8S_AWAIT_ALL=true` will cause Pulumi to automatically wait for _all_ resources to become ready when creating or updating them -- including custom resources.
+
+If a resource appears to follow well-known conventions (like the "Ready" condition, among others) Pulumi will wait for it to become ready.
+If the resource doesn't appear to follow well-known conventions, Pulumi will maintain its existing behavior and assume the resource is ready immediately.
+
+(Generic readiness criteria are determined by the [cli-utils](https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus) SIG, and behavior is unchanged for any resources that Pulumi already waits for readiness.)
```

Comment:
`cli-utils` is a collection of Go libraries that is maintained by [SIG CLI](https://github.com/kubernetes/community/blob/master/sig-cli/README.md).